tap_zendesk/http.py

```diff
@@ -208,7 +210,126 @@ def get_offset_based(url, access_token, request_timeout, page_size, **kwargs):
         yield response_json
         next_url = response_json.get('next_page')
 
-def get_incremental_export(url, access_token, request_timeout, start_time):
+
+async def raise_for_error_for_async(response):
+    """
+    Error handling method which throws custom error. Class for each error defined above which extends `ZendeskError`.
+    """
+    response_json = {}
+    try:
+        response_json = await response.json()
+    except ContentTypeError as e:
+        LOGGER.warning("Error decoding response from API: %s", str(e))
+    except ValueError as e:
+        LOGGER.warning("Invalid response from API: %s", str(e))
```

Comment:
If we are just logging the error, it makes sense to catch both exceptions together. This simplifies the code and reduces redundancy. Also the logger will include the exception details, making it easier to trace the issue.
```suggestion
    try:
    response_json = await response.json()
except (ContentTypeError, ValueError) as e:
    LOGGER.warning("Error decoding response from API. Exception: %s", e, exc_info=True)
```

Reply:
Agree